xbmc/listproviders/DirectoryProvider.cpp

```diff
@@ -272,7 +274,16 @@ void CDirectoryProvider::OnEvent(const ADDON::AddonEvent& event)
         typeid(event) == typeid(ADDON::AddonEvents::Disabled) ||
         typeid(event) == typeid(ADDON::AddonEvents::InstalledChanged) ||
         typeid(event) == typeid(ADDON::AddonEvents::MetadataChanged))
-    m_updateState = INVALIDATED;
+      m_updateState = INVALIDATED;
+  }
+}
+
+void CDirectoryProvider::Notify(const Observable &obs, const ObservableMessage msg)
+{
+  if (URIUtils::IsProtocol(m_currentUrl, "pvr"))
+  {
+    if (msg == ObservableMessageManagerStateChanged && g_PVRManager.IsStarted())
```

Comment:
I'm just trying to keep this clean from any dependency that isn't strictly registering and not leaking in details that doesn't have anything to do with the vfs from those singletons. Why does the directory provider need to know, or care, about which state pvr manager is in? Addons have similar state change events (InstalledChanged, MetadataChanged) but it doesn't call back into addonmgr.